src/components/splashscreen/splashscreen.js

```diff
@@ -44,11 +75,53 @@ class Splashscreen extends React.Component {
       this.props.peers.options.address === network.address;
   }
 
+  // eslint-disable-next-line class-methods-use-this
+  showNetworkOptions() {
+    const showNetwork = this.props.settings && this.props.settings.showNetwork;
+    const params = parseSearchParams(this.props.history.location.search);
+    const showNetworkParam = params.showNetwork || params.shownetwork;
+
+    return showNetworkParam === 'true' || (showNetwork && showNetworkParam !== 'false');
+  }
+
+  changeNetwork(network) {
+    this.setState({ network });
+    this.props.settingsUpdated({ network });
+  }
+
+  getNetwork(chosenNetwork) {
+    const network = { ...getNetwork(chosenNetwork) };
+    if (chosenNetwork === networks.customNode.code) {
+      network.address = addHttp(this.state.address);
+    }
+    return network;
+  }
+
+  getNetworksList() {
+    this.networks = Object.keys(networks)
+      .filter(network => network !== 'default')
+      .map((network, index) => ({
+        label: i18next.t(networks[network].name),
+        value: index,
+      }));
+  }
+
   render() {
-    const { t } = this.props;
+    const { t, settingsUpdated, peers } = this.props;
+
     return (
       <React.Fragment>
-        <HeaderV2 dark={true} showSettings={true} />
+        <HeaderV2
+          dark={true}
+          showSettings={true}
+          validationError={this.state.validationError}
+          liskAPIClientSet={this.props.liskAPIClientSet}
+          address={peers.options.address}
+          networkList={this.networks}
+          selectedNetwork={peers.options.code || 0}
+          handleNetworkSelect={this.changeNetwork}
+          settingsUpdated={settingsUpdated}
+          showNetwork={this.showNetworkOptions()} />
```

Comment:
Why are all of these params duplicated in splashscreen and login page? The only param that I really see necessary to pass to `HeaderV2` is ` dark={true}`. Everything else should be inside `HeaderV2`.